sdv/datasets/demo.py

```diff
@@ -104,15 +105,18 @@ def _get_data(modality, output_folder_name, in_memory_directory):
     return data
 
 
-def _get_metadata(modality, output_folder_name, in_memory_directory):
-    metadata = MultiTableMetadata() if modality == 'multi_table' else SingleTableMetadata()
+def _get_metadata(output_folder_name, in_memory_directory, dataset_name):
+    metadata = Metadata()
     if output_folder_name:
         metadata_path = os.path.join(output_folder_name, METADATA_FILENAME)
-        metadata = metadata.load_from_json(metadata_path)
+        metadata = metadata.load_from_json(metadata_path, dataset_name)
 
     else:
-        metadict = json.loads(in_memory_directory['metadata_v1.json'])
-        metadata = metadata.load_from_dict(metadict)
+        metadata_path = 'metadata_v2.json'
+        if metadata_path not in in_memory_directory:
+            metadata_path = 'metadata_v1.json'
+        metadict = json.loads(in_memory_directory[metadata_path])
+        metadata = metadata.load_from_dict(metadict, dataset_name)
```

Comment:
why does it need the dataset_name now? All of the jsons should be in the new format so they should just be loadable

Reply:
Not necessary but if it's a single table, that becomes the name of the table unless we are fine with a default table name in the case of a conversion.

Reply:
Yep those are the two places that this code is only used in. I have updated all the public buckets so if we wanted to we could leave out option 1

Reply:
They still have the old one right? I think we need to support people on old versions of sdv

Reply:
Not sure if people save our s3 bucket locally? Otherwise people will always download from our s3 public demodataset buckets but I would recommend to support it just in case.

Reply:
No I mean that their code wouldn't have these changes you made so it will require the old metadata file to be there with the old name

Reply:
Yes the existing metadata is still there so all older code will work as well.